classes/search/ArticleSearchDAO.inc.php

```diff
@@ -30,45 +30,46 @@ class ArticleSearchDAO extends SubmissionSearchDAO {
 	 */
 	public function getPhraseResults($journal, $phrase, $publishedFrom = null, $publishedTo = null, $type = null, $limit = 500) {
 		if (empty($phrase)) return array();
-
+	
 		$sqlFrom = '';
 		$sqlWhere = '';
 		$params = array();
-
+	
 		for ($i = 0, $count = count($phrase); $i < $count; $i++) {
-			if (!empty($sqlFrom)) {
-				$sqlFrom .= ', ';
-				$sqlWhere .= ' AND ';
-			}
+			$sqlFrom .= $i > 0 ? ', ' : '';
 			$sqlFrom .= 'submission_search_object_keywords o'.$i.' NATURAL JOIN submission_search_keyword_list k'.$i;
-			if (strstr($phrase[$i], '%') === false) $sqlWhere .= 'k'.$i.'.keyword_text = ?';
-			else $sqlWhere .= 'k'.$i.'.keyword_text LIKE ?';
-			if ($i > 0) $sqlWhere .= ' AND o0.object_id = o'.$i.'.object_id AND o0.pos+'.$i.' = o'.$i.'.pos';
-
+			$sqlWhere .= $i > 0 ? ' AND ' : '';
+			$sqlWhere .= 'k'.$i.'.keyword_text LIKE ?';
 			$params[] = $phrase[$i];
+			if ($i > 0) {
+				$sqlWhere .= ' AND o0.object_id = o'.$i.'.object_id AND o0.pos+'.$i.' = o'.$i.'.pos';
+			}
 		}
-
-		if (!empty($type)) {
-			$sqlWhere .= ' AND (o.type & ?) != 0';
+	
+		if (!empty($type) && $type == SUBMISSION_SEARCH_AUTHOR) {
```

Comment:
The previous code was incorrect (or perhaps outdated), assuming that we would a bitmask, but I didn't understand why you've limited the type to `SUBMISSION_SEARCH_AUTHOR`.